File: MyApplication2/common/src/main/java/com/example/common/base/BaseActivity.java
package com.example.common.base;

import android.os.Bundle;
import android.util.Log;

import com.example.common.utils.Cons;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

/**
 * 项目父Activity
 */
public class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Log.e(Cons.TAG, "common/BaseActivity");
    }
}

File: MyApplication2/app/src/main/java/com/example/mengzhi/lifecycletesting/view/BaseActivity.kt
package com.example.mengzhi.lifecycletesting.view

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.mengzhi.lifecycletesting.presenter.BasePresenter

abstract class BaseActivity<V : IBaseView, T : BasePresenter<V>> : AppCompatActivity() {
    protected var presenter: T? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        //由activity选择一个表示层
        presenter = createPresenter()
        presenter?.attachView(this as V)
        registerSDK()
        init()
    }

    protected abstract fun createPresenter(): T
    protected fun init() {}
    protected fun registerSDK() {}
    protected fun unRegisterSDK() {}
    override fun onDestroy() {
        super.onDestroy()
        presenter?.detachView()
        unRegisterSDK()
    }
}